spyder/workers/updates.py

```diff
@@ -166,24 +166,20 @@ def start(self):
             error_msg = HTTP_ERROR_MSG.format(status_code=page.status_code)
             logger.debug(err, stack_info=True)
         except Exception as err:
-            error = traceback.format_exc()
-            formatted_error = error.replace('\n', '<br>').replace(' ', '&nbsp;')
-
-            error_msg = _(
-                'It was not possible to check for Spyder updates due to the '
-                'following error:'
-                '<br><br>'
-                '<tt>{}</tt>'
-            ).format(formatted_error)
+            # Only log the error when it's a generic one because we can't give
+            # users proper feedback on how to address it. Otherwise we'd show
+            # a long traceback that most probably would be incomprehensible to
+            # them.
```

Comment:
Shouldn't we still show a message that the check for update failed then? Maybe setting the error message (`error_msg`) to be something more generic like `Unable to check for updates` or even, reusing the previous message, `It was not possible to check for Spyder updates`. If I'm understanding correctly, what this change does is to not show any dialog when a check for updates fails due to exceptions we don't know how to handle, right? Could that not end up making users think that that there is no update available instead of making clear that a check was not possible to be done? Following this understanding, in case there is actually an update, users then most probably end up falsely believing they are running an up to date version of Spyder, no @ccordoba12 ?

Reply:
> If I'm understanding correctly, what this change does is to not show any dialog when a check for updates fails due to exceptions we don't know how to handle, right?

Correct.

> Could that not end up making users think that that there is no update available instead of making clear that a check was not possible to be done?

That's true but up to a certain point. The thing is there are two problems with the current approach:

1. Users can't do anything on their side to fix a generic exception (like checking if their internet connection is working). Therefore, I think it's annoying to show them those errors because, as I said above, a generic traceback would not be easy to grasp for them.
    And even showing a message like `It was not possible to check for Spyder updates` is not that useful because (again) users can't take action to be correctly notified about updates.
2. Most probably a generic exception is due to an internal Spyder error. So, we need to connect this to our error report dialog. But since the code related to updates diverged significantly between 5.x and master, I think it's better to leave this for master.

> Following this understanding, in case there is actually an update, users then most probably end up falsely believing they are running an up to date version of Spyder, no @ccordoba12 ?

Yep, but giving users feedback about an available update is a benefit we're providing for them. If that benefit turns into an annoyance they can't deal with, then it's not useful anymore.

----

On an additional note, my rationale to do this change is because I received a long traceback about this when checking for updates with the Linux installer and found it really annoying and unnecessary.

Reply:
> Most probably a generic exception is due to an internal Spyder error. So, we need to connect this to our error report dialog. But since the code related to updates diverged significantly between 5.x and master, I think it's better to leave this for master.

Should an issue be open for that or maybe it already exists?

Reply:
There's no issue about it, so please open one. That way we won't forget to address it for Spyder 6.

Reply:
See #21882